pkg/irreparabledb/irreparabledb_test.go

```diff
@@ -0,0 +1,63 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package irreparabledb
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"storj.io/storj/internal/testcontext"
+)
+
+func TestIrreparableRepository(t *testing.T) {
+	//testing variables
+	SegmentInfo := RemoteSegmentInfo{
+		SegmentPath:        []byte("IamSegmentkeyinfo"),
+		SegmentDetail:      []byte("IamSegmentdetailinfo"),
+		LostPiecesCount:    int64(10),
+		RepairUnixSec:      time.Now().Unix(),
+		RepairAttemptCount: int64(10),
+	}
+
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	// creating in-memory db and opening connection
+	irrdb, err := New("file::memory:?mode=memory&cache=shared")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer ctx.Check(irrdb.db.Close)
+
+	t.Run("New Entry", func(t *testing.T) {
+		segmentInfo := &SegmentInfo
+		err = irrdb.IncrementRepairAttempts(ctx, segmentInfo)
+		assert.NoError(t, err)
+	})
+
+	t.Run("Increment the already existing entry", func(t *testing.T) {
+		segmentInfo := &SegmentInfo
+		err = irrdb.IncrementRepairAttempts(ctx, segmentInfo)
+		assert.NoError(t, err)
+		segmentInfo.RepairAttemptCount++
+
+		dbxInfo, err := irrdb.Get(ctx, segmentInfo.SegmentPath)
+		assert.NoError(t, err)
+		assert.EqualValues(t, segmentInfo.SegmentPath, dbxInfo.SegmentPath)
```

Comment:
You can compare the whole struct at once with `assert.Equal` instead of comparing one by one.

Reply:
done